aws/rust-runtime/aws-credential-types/src/cache/lazy_caching.rs

```diff
@@ -120,36 +115,25 @@ mod builder {
     use std::sync::Arc;
     use std::time::Duration;
 
+    use crate::cache::{CredentialsCache, Inner};
     use crate::provider::ProvideCredentials;
     use aws_smithy_async::rt::sleep::{default_async_sleep, AsyncSleep};
 
     use super::TimeSource;
     use super::{
-        LazyCachingCredentialsProvider, DEFAULT_BUFFER_TIME, DEFAULT_CREDENTIAL_EXPIRATION,
+        LazyCredentialsCache, DEFAULT_BUFFER_TIME, DEFAULT_CREDENTIAL_EXPIRATION,
         DEFAULT_LOAD_TIMEOUT,
     };
 
-    /// Builder for constructing a [`LazyCachingCredentialsProvider`].
-    ///
-    /// # Examples
-    ///
-    /// ```no_run
-    /// use aws_credential_types::Credentials;
-    /// use aws_credential_types::credential_fn::provide_credentials_fn;
-    /// use aws_credential_types::lazy_caching::LazyCachingCredentialsProvider;
+    /// Builder for constructing a `LazyCredentialsCache`.
     ///
-    /// let provider = LazyCachingCredentialsProvider::builder()
-    ///     .load(provide_credentials_fn(|| async {
-    ///         // An async process to retrieve credentials would go here:
-    ///         Ok(Credentials::new("example", "example", None, None, "my_provider_name"))
-    ///     }))
-    ///     .build();
-    /// ```
-    #[derive(Debug, Default)]
+    /// Callers outside of this crate cannot call `build` directly. They can instead call
+    /// `into_credentials_cache` to obtain a [`CredentialsCache`]. Its `create_cache` then calls
```

Comment:
Does the `CredentialCache` link here work? I can't remember if doc.rs handles links containing backticks correctly.

Reply:
Yes, confirmed it works.